example/src/main/java/com/vimeo/example/MainActivity.kt

```diff
@@ -0,0 +1,129 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.vimeo.example
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.vimeo.networking2.*
+import com.vimeo.networking2.config.VimeoApiConfiguration
+import kotlinx.android.synthetic.main.activity_main.*
+
+/**
+ * The sample activity in which we create an [Authenticator] instance to authenticate with the server.
+ */
+class MainActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        // Construct a configuration instance.
+        val configuration = VimeoApiConfiguration.Builder(CLIENT_ID, CLIENT_SECRET, listOf(ScopeType.PUBLIC))
+            .withCodeGrantRedirectUrl("https://example.com")
+            .build()
+
+        // Initialize the authenticator instance with the configuration and the api client.
+        val authenticator = Authenticator(configuration)
+        val apiClient = VimeoApiClient(configuration, authenticator)
+
+        // If we are handling a code grant redirect, this URI will be present, otherwise it's a normal app start.
+        val redirectUri = intent?.data
+
+        if (redirectUri != null) {
+            // Log in using the code grant URI provided to us by the redirect.
+            authenticator.authenticateWithCodeGrant(redirectUri.toString(), vimeoCallback(
+                onSuccess = {
+                    toast("Successfully logged in.")
+                    loginStatus.text = authenticator.currentAccount.asLoginStatus()
+                },
+                onError = { toast("Unable to log in.") }
+            ))
+        }
+
+        loginStatus.text = authenticator.currentAccount.asLoginStatus()
+
+        clientCredentials.setOnClickListener {
+            // Obtain logged out credentials which can be used to make logged out requests.
+            authenticator.clientCredentials(vimeoCallback(
+                onSuccess = {
+                    toast("Obtained client credentials.")
+                    loginStatus.text = authenticator.currentAccount.asLoginStatus()
+                },
+                onError = { toast("Couldn't obtain client credentials") }
```

Comment:
Now that the `MainActivity` explicitly calls this out in the documentation at the top, this is no longer necessary?

Reply:
I think it is still a helpful thing to have. People will make mistakes regardless of the docs (many never read them) and this would be a quick way to point them in the right direction.

Reply:
Since this isn't the only request that can fail due to the id and secret not being set, I am just going to add a dialog at the beginning of `onCreate` that stops them in their tracks.

Reply:
Great idea!